torch/distributions/transforms.py

```diff
@@ -600,6 +601,26 @@ def log_abs_det_jacobian(self, x, y):
         return -F.softplus(-x) - F.softplus(x)
 
 
+class SoftplusTransform(Transform):
+    r"""
+    Transform via the mapping :math:`\text{Softplus}(x) = \log(1 + \exp(x))`.
+    """
+    domain = constraints.real
+    codomain = constraints.positive
+
+    def __eq__(self, other):
+        return isinstance(other, SoftplusTransform)
+
+    def _call(self, x):
+        return x.exp().log1p()
+
+    def _inverse(self, y):
+        return y.expm1().log()
```

Comment:
Probably a more stable version will be:
```
(-y).expm1().neg().log() - y
```
The point is if possible, we might avoid using `exp(x)` for moderate `x`. Here we know that `y` is positive, so `-y` won't be a large number

Reply:
Since [`torch.nn.functional.softplus`](https://pytorch.org/docs/master/nn.functional.html#torch.nn.functional.softplus) reverts to a linear function when `x` is above `20` (default threshold), this won't pass the tests - alternatively, I could use the `softplus` for `log_abs_det_jacobian` and `_call` as you suggested and use this here:
```
torch.where(y > 20, y, y.expm1().log())
```
(will `torch.where` make it slower?)

Reply:
Sorry, my mistake, the correct formula should be `(-y).expm1().neg().log() + y`